paimon-core/src/main/java/org/apache/paimon/table/FileStoreTableFactory.java

```diff
@@ -36,13 +38,19 @@
 
 /** Factory to create {@link FileStoreTable}. */
 public class FileStoreTableFactory {
-
     public static FileStoreTable create(CatalogContext context) {
+        boolean resolvingFileIOEnabled =
+                context.options().get(CatalogOptions.RESOLVING_FILEIO_ENABLED);
         FileIO fileIO;
-        try {
-            fileIO = FileIO.get(CoreOptions.path(context.options()), context);
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+        if (resolvingFileIOEnabled) {
```

Comment:
Can we modify this in `FileIO.get`? And we can remove this option in`ResolvingFileIO` to avoid infinite recursion